### DISCUSSION TITLE
Unavailable daily energy sensors

### DISCUSSION THREAD

**Discussion by dmode9000** [slipx06/sunsynk-power-flow-card#445] [General]
Tried to use sunsynk-power-flow-card with PowMr invertor [according to the example](https://slipx06.github.io/sunsynk-power-flow-card/examples/powmr.html), but daily/total energy sensors are unavailable for PV and battery. How to fix this?
![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/111885298/674f4fbb-a122-4ab2-bb5a-a29fbbbb5d69)

**Comment by slipx06** [comment]
Have you checked those sensors in developer tools?

**Comment by dmode9000** [comment]
Which specific check do you mean? If it's just a search for such an entity, then yes, the search does not find such an entity. 
![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/111885298/eeb52812-4e7e-485c-bb01-b8ce42d15526)
![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/111885298/f094dc4d-e39f-4657-bd5a-5f58a1ab344f)
![image](https://github.com/slipx06/sunsynk-power-flow-card/assets/111885298/75e09eb8-b78e-4c5c-9bd9-69f0e8433a71)


Moreover, a search for the key "pv_yield_daily" or "discharge_daily" in the integration files at https://github.com/odya/esphome-powmr-hybrid-inverter does not return any mentions. So I don't understand where this sensor was supposed to be initialized.

Considering that the results of these sensors are simply the product of voltage and amperage, I could assume that these sensors can be created manually. However, since there is no mention of this in the documentation, and you mentioned on the forum that this sensor should be included in odya's integration, [I asked him a question](https://github.com/odya/esphome-powmr-hybrid-inverter/issues/43) but have not yet received a response.

**Reply by Roving-Ronin** [reply]
@dmode9000 the gent who developed the PowerMr inverter modbis integration has been provided the base example and suggested that he map out the sensors fully, but silence. IIRC I've done a PR with a LOT of updates for him and I believe its just sitting there, even though it would add functions / fix up stuff.

Update I did was at: https://github.com/Roving-Ronin/esphome-powmr-hybrid-inverter

BTW check your PowerMr inverter, as a mate got one of them for his caravan intending to run PV + Battery + Grid and was having problems in his house when the caravan was plugged in. Worked out that the inverter, which should NOT be able to export back to the grid was in fact leaking power back into the grid. When questioning the reseller / importer about this, was informed "it's a known issue" (read 110% illegal for use, at least in Australia). Had to rip it out and toss the unit.

In regards the issues you point to above, re daily production / battery charge and discharge etc.. you could have a look at the Huawei example yaml files, that can give you some examples of how to use the utility_meter (or even better then energy_meter integration, that also allows tracking $ in/out) to create the utility meters for daily/weekly etc to track and publish these sensors your after.

From memory I wasn't able to finalise it, as given the power leak issue the friend I was assisting to get this up and running, given he had to rip out the PowerMr meant I wasn't able to the PZEM sensors wired into an ESP32 and CT clamps placed around the battery discharge / discharge wires etc... so I couldn't have a running PowerMr with functioning sensors for everything to test and finalise the code.

If you have a fully setup PowerMr with PZEM + CT's (and DHT or better BME280 temp/humidity/pressure) sensors, that you could give me remote access too (Fixed IP my end) I could see if I could assist you to complete, when I get a moment.

**Comment by dmode9000** [comment]
Thank you for your willingness to help. I am currently in the process of switching to a different brand of inverter, so I see that the amount of effort required to solve this problem is not worth it.
